### PR TITLE
add manual contrib release step

### PR DESCRIPTION
**Pull request by timosachsenberg**
Key idea here is, as long as we use the contrib on e.g. windows, enable a simple way to get built artefacts and use them in the OpenMS/OpenMS or other actions. Right now, we rely on jenkins and the buildarchive here. This complicates things a lot as the contrib e.g. needs to be built for different visual studio versions (see https://abibuilder.cs.uni-tuebingen.de/archive/openms/contrib/windows/x64/).

Things I am unsure:
- how to select a proper tag that allows us to download and use the contrib.tar.gz from the OpenMS/OpenMS actions?

### CODE REVIEW

**Comment by jpfeuffer** [comment]
With artefacts v4 you can just download from the latest job of master in that repo.

**Comment by timosachsenberg** [comment]
k but general  artifacts are not preserved for a long time. Would a release not be more appropriate?

**Comment by jpfeuffer** [comment]
true. forgot. release is better

**Review by poshul — APPROVED** [review]

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -66,8 +68,35 @@ jobs:
...
+      run: |
+        git config --global user.name 'GitHub Actions'
+        git config --global user.email 'actions@github.com'
+
+    - name: Create and Push Tag
+      if: github.event_name == 'workflow_dispatch'
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        TAG_NAME="release-$(date +'%Y%m%d%H%M%S')"
```
Comment (timosachsenberg) [inline .github/workflows/main.yml:82]: not sure how to create a good tag name here or how to properly name files:
Currently I use:
./contrib-build/contrib_build-${{runner.os}}.tar.gz
for the zip.
Note that we need a way to download the proper one from the github actions in OpenMS/OpenMS

Comment (poshul) [reply .github/workflows/main.yml:82]: I don't see the naming here as especially important, we generate the tags that we use publicly on running the release action, so these are just for internal use right? The way I see it we can just have the OpenMS/OpenMS action grab the latest tag, and use the contrib file that corresponds to its own OS. Or am I missing something?

Comment (timosachsenberg) [reply .github/workflows/main.yml:82]: yeah maybe it is enough to just distinguish by os (and hope that the runner are all the same).
On jenkins we observed a lot of minor changes between XCode and VSCode versions that required individual contrib packages (each one was in a different folder on abibuilder)
We might just try it and then get the release with something like:
```
name: Fetch Latest Release

on:
  workflow_dispatch:

jobs:
  fetch-release:
    runs-on: ubuntu-latest
    steps:
      - name: Fetch latest release from another repository
        id: get-latest-release
        uses: actions/github-script@v6
        with:
          github-token: ${{ secrets.GH_TOKEN }}
          script: |
            const repoOwner = 'other-repo-owner';  // Replace with the owner of the repository
            const repoName = 'other-repo-name';    // Replace with the repository name
            const response = await github.repos.getLatestRelease({
              owner: repoOwner,
              repo: repoName
            });
            return response.data;

      - name: Use the release data
        run: |
          echo "The latest release tag is ${{ steps.get-latest-release.outputs.tag_name }}"
          echo "Download URL is ${{ steps.get-latest-release.outputs.zipball_url }}"
```
and then just extract / use the one matching the OS?
No idea if this works :-P

**Comment by timosachsenberg** [comment]
k YOLO. easy to revert ...

**Comment by timosachsenberg** [comment]
@JeeH-K WIP but we are currently trying to create the contrib as release on GitHub. If it works, we can use that instead of the one created by jenkins

**Pull request merged**